src/zarrzipan/cli.py

```diff
@@ -0,0 +1,335 @@
+    type=click.Path(dir_okay=False, path_type=Path),
+)
+@click.option(
+    '--output-dir',
+    default=OUTPUT_DIR,
+    help=(
+        'Directory to put output into. If not specified '
+        f"'{OUTPUT_DIR}' will be used unless output-file is specified"
+    ),
+    type=click.Path(file_okay=False, path_type=Path),
+)
+@click.option(
+    '--hide',
+    '-h',
+    is_flag=True,
+    help='Do not show the table.',
+)
+def run(
+    ctx,
+    config_file: Path,
+    data_dir: Path = DATA_DIR,
+    output_file: Path | None = None,
+    output_dir: Path = OUTPUT_DIR,
+    hide: bool = False,
+) -> None:
+    """Run the jobs in the config file fetching data if needed"""
+
+    config = Config.from_file(config_file)
+
+    if not output_file:
+        output_file = _get_output_filepath(output_dir)
+
+    results = []
+    for job in track(config.jobs, description='Running jobs...'):
+        local_path = _fetch_array(data_dir, job.array['name'], job.array['href'])
+
+        # Open the dataset from local with xarray
+        ds = xr.open_dataset(local_path, chunks=None)
+
+        # TODO: Add handling for when there is more than one variable
+        if len(ds.keys()) == 1:
+            data_array = next(v.squeeze() for v in ds.values())
+        else:
+            raise ValueError('More than one variable in dataset')
+
+        if job.array['slice'] is not None:
+            data_array = data_array.isel(
+                {k: slice(*v) for k, v in job.array['slice'].items()},
+            )
+
+        if ctx.obj['v']:
+            click.echo(f'Array has shape: {data_array.shape}')
+
+        pipelines = [
+            CodecPipeline(name=pipeline['name'], codec_configs=pipeline['steps'])
+            for pipeline in job.pipelines
+        ]
+
+        for chunk_shape in job.chunk_shapes:
+            dataset = BenchmarkInput(
+                name=job.array['name'],
+                array=data_array.values,
+                chunk_shape=tuple(chunk_shape) if chunk_shape else data_array.shape,
+            )
+
+            if ctx.obj['v']:
+                click.echo(
+                    f'Running benchmark comparison with chunk shape: {chunk_shape} ...',
+                )
+
+            run_results = asyncio.run(
+                run_comparison(
+                    datasets=[dataset],
+                    pipelines=pipelines,
+                    iterations=job.iterations,
+                ),
+            )
+
+            with output_file.open('a') as f:
+                for result in run_results.to_ndjson():
+                    f.write(result)
+                    f.write('\n')
+
+            results.extend(run_results.to_dicts())
+
+    if not hide:
+        table = _create_table(results)
+
+        # Print the table
+        console = Console()
+        console.print(table)
+
+
+@cli.command()
+@click.argument(
+    'output_file',
+    type=click.Path(exists=True, dir_okay=False, path_type=Path),
+)
+def show(output_file: Path) -> None:
```

Comment:
I wonder if `render` or something like that would be a more descriptive name for this operation.

Again, this is a place where I would push for providing the data to format via stdin instead of via an option/argument, to facilitate the pipeline-like operation I suggested above.